### PR TITLE
Implemented display: inline-flex

### LINKED ISSUE #14685: Implement display: inline-flex

### PR DESCRIPTION
**Pull request by shinglyu**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #14685 (github issue number if applicable).

- [x] There are tests for these changes 

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14978)

### CODE REVIEW

File: components/layout/construct.rs [hunk 1]
```diff
@@ -1316,6 +1316,39 @@ impl<'a, ConcreteThreadSafeLayoutNode: ThreadSafeLayoutNode>
         self.build_flow_for_block_like(flow, node)
     }
 
+    /// Builds a fragment for a node with 'display: inline-flex'.
+    // Shared code with build_fragment_for_inline_block
+    fn build_fragment_for_inline_flex(&mut self, node: &ConcreteThreadSafeLayoutNode)
```
Comment (shinglyu) [inline components/layout/construct.rs]: This function should probably reuse `build_fragment_for_inline_block`, but I'm not sure if we want to have a `build_fragment_for_inline_block_or_inline_flex` and pass a flag into it, or should we have two functions and extract the shared part into a helper function.

**Comment by shinglyu** [comment]
cc @stshine

**Comment by shinglyu** [comment]
r? anyone? 
@highfive where are you?

**Comment by stshine** [comment]
r? @mbrubeck or @pcwalton or @notriddle 
I am busy with my job, will try to look later.

File: components/layout/inline.rs [hunk 2]
```diff
@@ -1154,6 +1155,7 @@ impl InlineFlow {
                 }
                 (display::T::inline, vertical_align::T::bottom) |
                 (display::T::block, vertical_align::T::bottom) |
+                (display::T::inline_flex, vertical_align::T::top) |
```
Comment (mbrubeck) [inline components/layout/inline.rs]: `top` should be `bottom` here

**Review by mbrubeck — CHANGES_REQUESTED** [review]

File: components/layout/construct.rs (hunk 1 above)
Comment (stshine) [inline components/layout/construct.rs]: Yeah, I agree add a flag should be a better solution.

Comment (stshine) [inline components/layout/construct.rs]: Yeah, I agree add a flag should be a better solution.

File: components/style/properties/properties.mako.rs [hunk 3]
```diff
@@ -1899,7 +1899,8 @@ pub fn apply_declarations<'a, F, I>(viewport_size: Size2D<Au>,
         % if product == "gecko":
         computed_values::display::T::grid |
```
Comment (stshine) [inline components/style/properties/properties.mako.rs]: Why not also fix `inline-grid` for stylo?

File: tests/wpt/metadata-css/css-flexbox-1_dev/html/flex-flexitem-percentage-prescation.htm.ini [hunk 4]
```diff
@@ -1,4 +0,0 @@
-[flex-flexitem-percentage-prescation.htm]
```
Comment (stshine) [inline tests/wpt/metadata-css/css-flexbox-1_dev/html/flex-flexitem-percentage-prescation.htm.ini]: This test actually does not pass.

**Review by stshine — COMMENTED** [review]
Disclaimer: I do not have access as a reviewer.

File: components/layout/construct.rs (hunk 1 above)
Comment (KiChjang) [inline components/layout/construct.rs]: Also, small nit: the `Shared code` comment needs another slash in front of it to be a doc comment.

File: components/style/properties/properties.mako.rs (hunk 3 above)
Comment (shinglyu) [inline components/style/properties/properties.mako.rs]: fixed!

**Comment by shinglyu** [comment]
Let me know when it's ready for squashing.

**Comment by stshine** [comment]
Looks good to me, nice work :-)
Your need an actual reviewer for the review process.

**Comment by notriddle** [comment]
Squash this, then I'll r+ it.

**Comment by shinglyu** [comment]
@notriddle Squahed. Thank you!

**Comment by stshine** [comment]
@bors-servo retry

internal compiler error

**Pull request merged**